osa_cli_releases/releasing.py

```diff
@@ -259,24 +261,33 @@ def update_ansible_role_requirements_file(
         # Freeze sha by checking its trackbranch value
         # Do not freeze sha if trackbranch is None
         if trackbranch:
-            # Unfreeze on master, not bump
-            if branchname == "master" and not milestone_freeze:
-                print("Unfreeze master role")
-                role["version"] = trackbranch
-            # Freeze or Bump
-            else:
-                role["version"] = get_sha_from_ref(role["src"], trackbranch)
-                print("Bumped role %s to sha %s" % (role["name"], role["version"]))
-
-        # Copy the release notes `Also handle the release notes
-        # If frozen, no need to copy release notes.
-        if copyreleasenotes and trackbranch:
-            print("Cloning and copying %s's release notes" % role["name"])
-            _, role_path = clone_role(
-                role["src"], branchname, clone_root_path, depth="1"
-            )
-            copy_role_releasenotes(role_path, "./")
-            shutil.rmtree(role_path)
+            try:
+                role_repo = clone_role(
+                   role["src"], trackbranch, clone_root_path, depth="1"
+                )
+                # Unfreeze on master, not bump
+                if branchname == "master" and not milestone_freeze:
+                    print("Unfreeze master role")
+                    role["version"] = trackbranch
+                # Freeze or Bump
+                else:
+                    role_head = role_repo.head()
+                    role["version"] = role_head.decode()
+                    print("Bumped role %s to sha %s" % (role["name"], role["version"]))
+
+                    if shallow_since:
+                        head_timestamp = role_repo[role_head].commit_time
+                        head_datetime = datetime.fromtimestamp(head_timestamp)
+                        role["shallow_since"] = head_datetime.strftime('%Y-%m-%d')
+
+                # Copy the release notes `Also handle the release notes
+                # If frozen, no need to copy release notes.
+                if copyreleasenotes:
```

Comment:
you've changed logic there: You clone the repo at all times. it's not necessary to clone in all functions, but I like the simplification.

However, next to that, you removed trackbranch. How do you ensure that you don't copy the release notes of new changes while freezing/in the frozen period?

Reply:
I removed branchname not trackbranch? Or you mean condition? If yes, then I guess it's useless because we're still under `if trackbranch` section.

But I guess I just haven't realized how we're protecting indeed. Maybe it's worth adding here `and not milestone_freeze` as well?

Reply:
Yes, I misread the indentation on previous review.